tests/components/datadog/test_config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for the Datadog config flow."""
+
+from unittest.mock import patch
+
+from homeassistant.components import datadog
+from homeassistant.components.datadog.config_flow import DatadogConfigFlow
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .common import MOCK_CONFIG
+
+from tests.common import MockConfigEntry
+
+
+async def test_user_flow_success(hass: HomeAssistant) -> None:
+    """Test user-initiated config flow."""
+    with patch(
+        "homeassistant.components.datadog.config_flow.validate_datadog_connection",
```

Comment:
```suggestion
        "homeassistant.components.datadog.config_flow.DogStatsd.increment",
```
We should mock the library and not the internal function in `config_flow.py`